feat: envtest setup via target with docs (v3 only) (#1626)
kubernetes-sigs/kubebuilder: docs/book/src/reference/envtest.md

Logs from the test runs are prefixed with `test-env`.

### Configuring your test control plane

> The controller-runtime’s [envtest](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/envtest) framework requires `kubectl`, `kube-apiserver`, and `etcd` binaries to simulate the API portions of a real cluster. 

Reviewer: ```suggestion
controller-runtime’s [envtest](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/envtest) framework requires `kubectl`, `kube-apiserver`, and `etcd` binaries be present locally to simulate the API portions of a real cluster. 
```

> For projects built with `v3+`, these binaries and configuration is done via the `test` makefile target. It will create the `testbin/` directory where the binaries will be downloaded.  

Reviewer: ```suggestion
For projects built with plugin v3+ (see your PROJECT file's `layout` key), the `make test` command will install these binaries to the `testbin/` directory and use them when running tests that use `envtest`.
```

> You can use environment variables and/or flags to specify the `kubectl`,`api-server` and `etcd` setup within your integration tests.

Reviewer: Link needed.

Author: Hi @estroz, 

I did not get what link should be added here? This text was not changed at all and it is added because bellow it has the envvar/flags details.

| `KUBEBUILDER_ATTACH_CONTROL_PLANE_OUTPUT` | boolean | Set to `true` to attach the control plane's stdout and stderr to os.Stdout and os.Stderr. This can be useful when debugging test failures, as output will include output from the control plane. |

See that the `test` makefile target will ensure that all is properly setup when you are using it. However, if you would like to run the tests without use the Makefile targets, for example via an IDE, then you can set the environment variables directly in the code of your `suite_test.go`:

```go 
var _ = BeforeSuite(func(done Done) {
	Expect(os.Setenv("TEST_ASSET_KUBE_APISERVER", "../testbin/bin/kube-apiserver")).To(Succeed())
	Expect(os.Setenv("TEST_ASSET_ETCD", "../testbin/bin/etcd")).To(Succeed())
	Expect(os.Setenv("TEST_ASSET_KUBECTL", "../testbin/bin/kubectl")).To(Succeed())

	logf.SetLogger(zap.New(zap.WriteTo(GinkgoWriter), zap.UseDevMode(true)))
	testenv = &envtest.Environment{}

> 	var err error
> 	cfg, err = testenv.Start()

Reviewer 2: nit: how do you feel about using the following instead, it's slightly less verbose
```suggestion
	cfg, err := testenv.Start()
```

Author: good cather.
